src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/ProblemMessageServiceTest.java

```diff
@@ -0,0 +1,104 @@
+        assertEquals(0, messages.size());
+        assertEquals(Collections.emptyList(), messages);
+    }
+
+    @Test
+    void should_return_unprocessable_when_case_id_is_invalid() {
+        CaseEventMessageEntity mockCaseEventMessageEntity = createMockCaseEventMessageEntity(
+            MessageState.UNPROCESSABLE);
+
+        when(caseEventMessageRepository.findProblemMessages(messageTimeLimit))
+            .thenReturn(Collections.singletonList(mockCaseEventMessageEntity));
+
+        when(caseEventMessageMapper.mapToCaseEventMessage(mockCaseEventMessageEntity))
+            .thenReturn(createMockCaseEventMessage(MessageState.UNPROCESSABLE));
+
+        List<CaseEventMessage> unprocessableMessages = problemMessageService
+            .findProblemMessages(JobName.FIND_PROBLEM_MESSAGES);
+        assertEquals(1, unprocessableMessages.size());
+        assertEquals(MESSAGE_ID, unprocessableMessages.get(0).getMessageId());
+        assertEquals(MessageState.UNPROCESSABLE, unprocessableMessages.get(0).getState());
+    }
+
+    @Test
+    void should_return_empty_list_when_no_ready_message_is_found() {
+        when(caseEventMessageRepository.findProblemMessages(messageTimeLimit)).thenReturn(Collections.emptyList());
+        List<CaseEventMessage> messages = problemMessageService.findProblemMessages(JobName.FIND_PROBLEM_MESSAGES);
+        assertEquals(0, messages.size());
+        assertEquals(Collections.emptyList(), messages);
+    }
+
+    @Test
+    void should_handle_ccd_case_event_asb_message_when_return_old_ready_messages() {
+        final CaseEventMessageEntity mockCaseEventMessageEntity = createMockCaseEventMessageEntity(MessageState.READY);
+        when(caseEventMessageRepository.findProblemMessages(messageTimeLimit))
+            .thenReturn(Collections.singletonList(mockCaseEventMessageEntity));
+        when(caseEventMessageMapper.mapToCaseEventMessage(mockCaseEventMessageEntity))
+            .thenReturn(createMockCaseEventMessage(MessageState.READY));
+        List<CaseEventMessage> readyMessages = problemMessageService.findProblemMessages(JobName.FIND_PROBLEM_MESSAGES);
+        assertEquals(1, readyMessages.size());
+        assertEquals(MESSAGE_ID, readyMessages.get(0).getMessageId());
+        assertEquals(MessageState.READY, readyMessages.get(0).getState());
+
+    }
+
+    private CaseEventMessageEntity createMockCaseEventMessageEntity(MessageState messageState) {
+        CaseEventMessageEntity caseEventMessageEntity = new CaseEventMessageEntity();
+        caseEventMessageEntity.setMessageId(MESSAGE_ID);
+        caseEventMessageEntity.setState(messageState);
+        return caseEventMessageEntity;
+    }
+
+    private CaseEventMessage createMockCaseEventMessage(MessageState messageState) {
+        return CaseEventMessage.builder()
+            .messageId(MESSAGE_ID)
+            .state(messageState)
+            .build();
```

Comment:
Think this is ok again but would be nicer if we had a flexible CaseEventMessageCreator with methods to create you a CreateEventMessage for different tests.
If they are specific to a test then the next dev will not see this and write the same thing again somewhere else.

Reply:
i will look into it.